cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -1031,23 +1027,42 @@ public CloudServiceInstance getServiceInstance(String serviceName, boolean requi
     }
 
     @Override
-    public CloudSpace getSpace(String spaceName) {
-        return getSpace(spaceName, true);
-    }
-
-    @Override
-    public CloudSpace getSpace(String spaceName, boolean required) {
-        Map<String, Object> resource = findSpaceResource(spaceName);
+    public CloudSpace getSpace(UUID spaceGuid) {
```

Comment:
Same here? Are there some methods which are hoping no exception to be thrown here?

Reply:
We do `restTemplate.getForObject()`, which throws an exception if the backend returns 404. As a result, this check is unreachable and essentially dead code.